### PR TITLE
Proposal: update eth/gnt balance on error only if some timeout has passed

### PR DESCRIPTION
**Pull request by mfranciszkiewicz**
This addresses the rapidly changing eth / gnt balance bug

### CODE REVIEW

File: golem/ethereum/client.py [hunk 1]
```diff
@@ -92,7 +92,7 @@ def get_balance(self, account, block=None):
             return self.web3.eth.getBalance(account, block)
         except ValueError as e:
             log.error("Ethereum RPC: {}".format(e))
```
Comment (chfast) [inline golem/ethereum/client.py:94]: What kind of error happens here? Using local or remote geth?

**Review by chfast — COMMENTED** [review]
I'm ok with this hotfix.

In longer run, we should merge ethereum.Node and ethereum.Client and handle geth quirks in more generic way there. I don't want to end up with a ton of different timeout params for communication with Ethereum.

File: golem/ethereum/paymentprocessor.py [hunk 2]
```diff
@@ -61,6 +61,9 @@ class PaymentProcessor(LoopingCallService):
     # TODO: Adjust this value later and add MAX_PAYMENTS limit.
     GAS_RESERVATION = 21000 + 1000 * 50000
 
+    # Time required to reset the current balance when errors occur
+    BALANCE_RESET_TIMEOUT = 10
```
Comment (Krigpl) [inline golem/ethereum/paymentprocessor.py]: I think we update the balances every 13 seconds or so. A value of 30 seconds might be more appropriate I guess.

**Review by Krigpl — APPROVED** [review]
Should be fine for now.

File: golem/ethereum/client.py (hunk 1 above)
Comment (mfranciszkiewicz) [reply golem/ethereum/client.py:94]: As far as I can tell, `ValueError`s are thrown when the json response contains an `error` property. This is specific to the HTTP provider though.

File: tests/golem/transactions/ethereum/test_paymentprocessor.py [hunk 3]
```diff
@@ -635,3 +635,14 @@ def test_no_gnt_available(self):
 
         q = Payment.create(subtask='F', payee=payee, value=value)
         assert not self.pp.add(q)
+
+    def test_balance_value(self):
```
Comment (etam) [inline tests/golem/transactions/ethereum/test_paymentprocessor.py:639]: you may want to use `freezegun` package to avoid any timing issues.

**Review by etam — APPROVED** [review]
As a hot-fix it looks OK.

**Pull request merged**